### ISSUE TITLE
[DetailsList] Horizontal scroll bar not available

### ISSUE THREAD

**Issue by flyingbee2012** [microsoft/fluentui#16715]
I generated a table using DetailsList where it has lots of columns. When I resize the window, the rest of the columns are clipped. So, I'd like to apply a horizontal scroll bar, but it seems there is a limitation to do so with DetailsList.  As a result, I am unable to see the all columns if I resize the window.

### Environment Information

- **Package version(s)**: DetailsList
- **Browser and OS versions**: Chrome, Windows 10

### Please provide a reproduction of the bug in a codepen:
https://codepen.io/flyingbee2012/pen/gOLOMbW

#### Actual behavior:

After applying "overflow: auto" in css, the horizontal scrollbar only applies to the column header instead of the whole list

#### Expected behavior:

The horizontal  scrollbar should apply to the whole list.

### Priorities and help requested:

Are you willing to submit a PR to fix? No

Requested priority: Normal

Products/sites affected: OWA

**Comment by rcbevans** [comment]
Your codepen running in Firefox on MacOS shows an issue where the header and body of the DetailsList are scrollable, but scroll independently and the body scrollbar doesn't seem to appear.

It is caused by the css:
```
div {
  overflow: auto;
}
```

Removing this the, header and content scroll together and the scrollbar appears just fine.

**Comment by flyingbee2012** [comment]
Thanks. After I removed this, I did see the horizontal scroll bar at bottom, but it only shows when I drag the vertical bar to the very bottom. Is there a way to show horizontal bar as long as the width exceeds the limit so that I don't need to scroll to the bottom in order to use it?

I increased the number of items to 100, as you can see, I cannot use the horizontal unless it reaches the bottom of the page
https://codepen.io/flyingbee2012/pen/gOLOMbW

**Comment by gouttierre** [comment]
@flyingbee2012 - thank you for the feedback. 

I see the problem you experiencing, if you open your [codepen](https://codepen.io/flyingbee2012/pen/gOLOMbW), start scrolling down the `DetailsList `and change the browser window size (smaller), the `DetailsList `horizontal bar only becomes visible once you reach the last row in the `List `and then it disappears as you scroll up.

![image](https://user-images.githubusercontent.com/78806055/110657830-736b1180-81c1-11eb-8691-eb68b820a629.png)

@khmakoto - Would you be able to if this is a regression or if this behavior is an issue? cc @smhigley

**Comment by flyingbee2012** [comment]
@gouttierre @layershifter Just checkin. I am wondering if you are looking into this issue. We have an accessibility bug in OWA of which the fix is dependent on this issue.  Kindly let me know if there is an ETA or workaround. Thanks!

**Comment by schutztj** [comment]
@khmakoto 
The issue here is that overflow-x is being applied on the .ms-DetailsList div. This div is sized based on the entire content of the list and not the virtualized viewport. Overflow-x needs to be set on a div that is sized based on the virtualized view and not the entire content to work properly.  Alternatively, there is no need to set overflow-x at all since the list does not virtualize in the horizontal plane and consumers could simply configure their overflow behavior on a parent container of the list.

![image](https://user-images.githubusercontent.com/1100754/149198743-fc94e096-3c2f-4336-a18b-75192f97657b.png)

**Comment by hunterboerner** [comment]
I've come across this issue as well. The horizontal scroll bar is only visible when you reach the bottom of the list. This seems like a bug.

**Comment by khmakoto** [comment]
@hunterboerner do you want me to reopen this issue?

**Comment by davidritch1996** [comment]
Chiming in here, same issue with horizontal bar only being visible when scrolled down. It would be helpful to reopen the ticket and fix this. We recently received an accessibility ticket related to this as well, as when zoomed in on the browser 400% it was practically impossible to scroll right and keep track of which list item is being looked at.

**Comment by timbellay** [comment]
While this seems like a low-medium severity accessibility bug, the concern here is that these types of bugs can easily kill feature adoption because they tend to make data rich apps painful to use to pull out insights.

The expected behavior is that one would see a horizontal scroll bar before reaching the end of the list (i.e. pinned to bottom on visible part of control) so that users can scroll the rows to see additional column information easily. 

Currently users with large lists have to 1) scroll to the bottom (could be 1000's of rows, or worse require pagination and loading), 2) guess how far to horizontal scroll (pertinent information in different columns may not fit on screen at same time) and then 3) vertically scroll back to the rows they need to scan for (could be hundreds of row).

**Comment by noumanfarooq566** [comment]
is any solution for this?

**Comment by Dissident26** [comment]
Not sure if it helps, but I've managed to deal with it by setting corresponding prop
![image](https://user-images.githubusercontent.com/60570678/230060526-9ae4e2fb-c229-49e8-8169-14d5dbf07e90.png)

**Comment by spmonahan** [comment]
> Not sure if it helps, but I've managed to deal with it by setting the corresponding prop ...

I believe this is the correct solution.

![constrain_mode](https://user-images.githubusercontent.com/93940821/230664336-7db312ff-5922-44c5-885e-7a4d6632e47a.png)

I'll update the docs to note this.
